admin/app/dfp/DataAgent.scala

```diff
@@ -10,24 +11,28 @@ trait DataAgent[K, V] extends Logging with implicits.Strings {
   private val initialCache: DataCache[K, V] = DataCache(Map.empty[K, V])
   private lazy val cache = AkkaAgent(initialCache)
 
+  def blockingOperations: BlockingOperations
+
   def loadFreshData(): Try[Map[K, V]]
 
   def refresh()(implicit executionContext: ExecutionContext): Future[DataCache[K, V]] = {
     log.info("Refreshing data cache")
     val start = System.currentTimeMillis
-    cache alterOff { oldCache =>
-      loadFreshData() match {
-        case Success(freshData) if freshData.nonEmpty =>
-          val duration = System.currentTimeMillis - start
-          log.info(s"Loading DFP data (${freshData.keys.size} items}) took $duration ms")
-          DataCache(freshData)
-        case Success(_) =>
-          log.error("No fresh data loaded so keeping old data")
-          oldCache
-        case Failure(e) =>
-          log.error("Loading of fresh data has failed.", e)
-          oldCache
-      }
+    blockingOperations.executeBlocking(loadFreshData()).map(freshIfExists(start))
+  }
+
+  private def freshIfExists(start: Long)(freshData: Try[Map[K, V]]): DataCache[K, V] = {
+    freshData match {
+      case Success(freshData) if freshData.nonEmpty =>
+        val duration = System.currentTimeMillis - start
+        log.info(s"Loading DFP data (${freshData.keys.size} items}) took $duration ms")
+        DataCache(freshData)
+      case Success(_) =>
+        log.error("No fresh data loaded so keeping old data")
+        cache.get
+      case Failure(e) =>
+        log.error("Loading of fresh data has failed.", e)
+        cache.get
     }
```

Comment:
The cache is not being updated with the new value here neither

Reply:
Fixed